### PR TITLE
Feat/dh shared pagination component

### PR DESCRIPTION
**Pull request by frontend-specialisten**
## Description

Shared pagination component.

## References

- #941 #1032

### CODE REVIEW

File: libs/dh/shared/ui-paginator/src/lib/dh-shared-ui-paginator.component.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    >
+    </mat-paginator>
+  `,
+})
+export class DhSharedUiPaginatorComponent implements OnDestroy {
+  @ViewChild(MatPaginator, { static: true }) private paginator!: MatPaginator;
+
+  @Input() length?: number;
+  @Input() pageSizeOptions: number[] = [50, 100, 150, 200, 250];
+  @Input() pageSize?: number;
```
Comment (ManBearTM) [inline libs/dh/shared/ui-paginator/src/lib/dh-shared-ui-paginator.component.ts:46]: Question: Should this have a default value as well?

File: libs/dh/shared/ui-paginator/src/lib/dh-shared-ui-paginator.component.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+  ariaLabel?: string;
+
+  private matPaginatorIntl = inject(MatPaginatorIntl);
+  private translocoService = inject(TranslocoService);
+  private destroy$ = new Subject<void>();
+
+  getPaginator = () => this.paginator;
```
Comment (ManBearTM) [inline libs/dh/shared/ui-paginator/src/lib/dh-shared-ui-paginator.component.ts:54]: Suggestion: Have you considered making this a getter function instead? The function call makes it seem like it has a non-trivial cost (fx. I would be hesitant to call this in ngOnChanges, even though it is basically free). Although `this.paginator.paginator` looks wrong, perhaps `this.paginator.instance`?

File: libs/dh/shared/ui-paginator/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+   ```ts
+   @ViewChild(DhSharedUiPaginatorComponent) paginator!: DhSharedUiPaginatorComponent;
+   ```
+
+4. Connect the data source with the paginator:
+
+   ```ts
+   ngAfterViewInit() {
+     this.dataSource.paginator = this.paginator.getPaginator();
```
Comment (ManBearTM) [inline libs/dh/shared/ui-paginator/README.md:25]: If you decide to change the function call as mentioned below, then don't forget to update the docs here 😄

File: libs/dh/shared/ui-paginator/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+| --------------- | ------------------------------------------------------------------------------------------------------------------------------------------ | -------------------------- |
+| length          | The length of the total number of items that are being paginated. **Note: using `MatTableDataSource` will overwrite this value**. `number` | `0`                        |
+| pageSizeOptions | The set of provided page size options to display to the user. `number[]`                                                                   | `[50, 100, 150, 200, 250]` |
+| pageSize        | Number of items to display on a page. `number`                                                                                             | `50`                       |
+
+## MatTableDataSource
+
+The built-in `MatTableDataSource` is designed for filtering, sorting and pagination of a **client-side** data array.
+For server-side filtering, sorting and pagination a custom data source is necessary [Read more.](https://blog.angular-university.io/angular-material-data-table)
```
Comment (ManBearTM) [inline libs/dh/shared/ui-paginator/README.md:40]: Praise: Love the docs!

File: libs/dh/shared/ui-paginator/src/lib/dh-shared-ui-paginator.component.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  @ViewChild(MatPaginator, { static: true }) private paginator!: MatPaginator;
+
+  @Input() length?: number;
+  @Input() pageSizeOptions: number[] = [50, 100, 150, 200, 250];
+  @Input() pageSize?: number;
+
+  ariaLabel?: string;
+
+  private matPaginatorIntl = inject(MatPaginatorIntl);
+  private translocoService = inject(TranslocoService);
```
Comment (ManBearTM) [inline libs/dh/shared/ui-paginator/src/lib/dh-shared-ui-paginator.component.ts:51]: Noob question: Is there any difference between this and constructor DI?

**Review by ManBearTM — COMMENTED** [review]
Stellar work, removes a lot of duplication! Only a few comments, just minor stuff.

File: libs/dh/shared/ui-paginator/src/lib/dh-shared-ui-paginator.component.ts (hunk 1 above)
Comment (frontend-specialisten) [reply libs/dh/shared/ui-paginator/src/lib/dh-shared-ui-paginator.component.ts:46]: Angular materials sets the `pagesize` to have a default value to 50, I will add it to make it explicit :) 
image

File: libs/dh/shared/ui-paginator/src/lib/dh-shared-ui-paginator.component.ts (hunk 5 above)
Comment (frontend-specialisten) [reply libs/dh/shared/ui-paginator/src/lib/dh-shared-ui-paginator.component.ts:51]: There shouldn't be any differences regarding the dependency injection itself. However, this syntax is not as "magic" as the constructor-based injection and actually, this also makes it possible to make reusable functions using dependency injections. Example:
image

Comment (ManBearTM) [reply libs/dh/shared/ui-paginator/src/lib/dh-shared-ui-paginator.component.ts:51]: I love it, will definitely use that more!

**Review by ManBearTM — APPROVED** [review]
💯

**Pull request merged**